packages/netsuite-adapter/src/query.ts

```diff
@@ -38,46 +59,62 @@ export type NetsuiteQuery = {
+  const invalidTypes = receivedTypes
+    .filter(recivedTypeName =>
+      !(existingTypes
+        .some(existTypeName => (checkTypeNameRegMatch({ name: recivedTypeName }, existTypeName)))))
 
-  const invalidRegexes = _.flatten(Object.values(parameters.types))
-    .concat(parameters.filePaths)
+  const invalidRegexes = idsRegexes
+    .concat(fileCabinet)
+    .concat(receivedTypes)
     .filter(reg => !regex.isValidRegex(reg))
 
-  if (invalidRegexes.length !== 0 || invalidTypes.size !== 0) {
+  if (invalidRegexes.length !== 0 || invalidTypes.length !== 0) {
     const invalidRegexesMessage = invalidRegexes.length !== 0 ? ` The following regular expressions are invalid: ${invalidRegexes}.` : ''
-    const invalidTypesMessage = invalidTypes.size !== 0 ? ` The following types do not exist: ${Array.from(invalidTypes)}.` : ''
-    const message = `received invalid ${FETCH_TARGET} input.${invalidRegexesMessage}${invalidTypesMessage}`
+    const invalidTypesMessage = invalidTypes.length !== 0 ? ` The following types or regular expressions do not match to existing types: ${invalidTypes}.` : ''
+    const message = `received invalid adapter config input.${invalidRegexesMessage}${invalidTypesMessage}`
     throw new Error(message)
   }
 }
 
 export const buildNetsuiteQuery = (
-  { types = {}, filePaths = [] }: Partial<NetsuiteQueryParameters>
-): NetsuiteQuery => {
-  const parameters = { types, filePaths }
-  return {
-    isTypeMatch: typeName => parameters.types[typeName] !== undefined,
-    areAllObjectsMatch: typeName => parameters.types[typeName] !== undefined && parameters.types[typeName].some(pattern => pattern === '.*'),
-    isObjectMatch: objectID => {
-      const regexes = parameters.types[objectID.type]
-      if (regexes === undefined) {
-        return false
-      }
-      return regexes.some(reg => new RegExp(`^${reg}$`).test(objectID.instanceId))
-    },
-    isFileMatch: filePath =>
-      parameters.filePaths.some(reg => new RegExp(`^${reg}$`).test(filePath)),
-    areSomeFilesMatch: () => parameters.filePaths.length !== 0,
-  }
-}
+  { types = [], fileCabinet = [] }: Partial<QueryParams>
+): NetsuiteQuery => ({
+  isTypeMatch: typeName => types
+    .some(type => checkTypeNameRegMatch(type, typeName)),
+  areAllObjectsMatch: typeName => _.flatten(types
+    .filter(type => checkTypeNameRegMatch(type, typeName))
+    .map(type => type.ids))
+    .some(id => (id === undefined || id === '.*')),
+  isObjectMatch: objectID => {
+    const relevantIds = types
+      .flatMap(type => (checkTypeNameRegMatch(type, objectID.type) ? type.ids : []))
+    const regexes = relevantIds
+      .map(id => (id === undefined ? '.*' : id))
+    return regexes.some(reg => new RegExp(`^${reg}$`).test(objectID.instanceId))
+  },
+  isFileMatch: filePath =>
+    fileCabinet.some(reg => new RegExp(`^${reg}$`).test(filePath)),
+  areSomeFilesMatch: () => fileCabinet.length !== 0,
+})
```

Comment:
Nothing here is critical, but I think that it contains more code reuse and some usages of the language internals instead of using lodash.
```suggestion
  export const buildNetsuiteQuery = (
  { types = [], fileCabinet = [] }: Partial<QueryParams>
): NetsuiteQuery => {
  const matchingTypes = (typeName: string): FetchTypeQueryParams[] =>
    types.filter(type => checkTypeNameRegMatch(type, typeName))

  const matchingTypesRegexes = (typeName: string): string[] =>
    matchingTypes(typeName)
      .flatMap(type => type.ids ?? ['.*'])

  return {
    isTypeMatch: typeName => matchingTypes(typeName).length > 0,
    areAllObjectsMatch: typeName =>
      matchingTypesRegexes(typeName)
        .some(id => id === '.*'),
    isObjectMatch: objectID =>
      matchingTypesRegexes(objectID.type)
        .some(reg => new RegExp(`^${reg}$`).test(objectID.instanceId)),
    isFileMatch: filePath =>
      fileCabinet.some(reg => new RegExp(`^${reg}$`).test(filePath)),
    areSomeFilesMatch: () => fileCabinet.length !== 0,
  }
}
```